packages/live-share-canvas/src/core/LiveCanvas.ts

```diff
@@ -775,20 +784,26 @@ export class LiveCanvas extends DataObject {
 
     private updateCursorPosition(
         clientId: string,
-        userInfo?: IUserInfo,
+        eventUserInfo?: IEventUserInfo,
         position?: IPoint
     ) {
-        if (this._inkingManager) {
-            if (position) {
-                const liveCursor = this.getCursor(clientId, userInfo);
-
-                const screenPosition =
-                    this._inkingManager.viewportToScreen(position);
-
-                liveCursor.setPosition(screenPosition);
-            } else {
-                this.removeCursor(clientId);
-            }
+        if (position) {
+            LiveShareClient.getClientInfo(clientId).then((clientInfo) => {
+                if (this._inkingManager) {
```

Comment:
Why not keep this as the first condition? It's cheap (I assume cheaper than `LiveShareClient.getClientInfo`) and necessary to run the code anyway.

Reply:
isn't there a thing in typescript where if using something that is possibly undefined in an async function, that you have to do the if check again or it will error out, since it could technically be reset to undefined before the async op completes?

Reply:
You might be right

Reply:
I'm only saying that ideally we don't do something expensive unless it's going to be useful, and here if `this._inkingManager` is undefined then `LiveShareClient.getClientInfo` will be called for nothing. Technically it's really not that big a deal because chances are `LiveCanvas` will always have an associated `InkingManager`, but still.

Reply:
discussed in a meeting, okay with changes, and getClientInfo is cheap to call